sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalysisContext.java

```diff
@@ -49,16 +49,16 @@ public ExpressionAnalysisContext() {
 
     public Function allocateFunction(FunctionInfo functionInfo, List<Symbol> arguments) {
         Function newFunction = new Function(functionInfo, arguments);
-        Function existingFunction = functionSymbols.get(newFunction);
-
-        if (existingFunction == null) {
-            functionSymbols.put(newFunction, newFunction);
-            hasAggregates = hasAggregates || functionInfo.type() == FunctionInfo.Type.AGGREGATE;
-
-            return newFunction;
-        } else {
-            return existingFunction;
+        if (functionInfo.isDeterministic()) {
+            Function existingFunction = deterministicFunctionSymbols.get(newFunction);
+            if (existingFunction == null) {
+                deterministicFunctionSymbols.put(newFunction, newFunction);
+                hasAggregates = hasAggregates || functionInfo.type() == FunctionInfo.Type.AGGREGATE;
```

Comment:
hasAggregates is not updated in case there is a nondeterministic aggregation function